### ISSUE TITLE
node-fetch high severity vulnerabilities

### ISSUE THREAD

**Issue by nadjin** [nestjs/nest#9001]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

review	node-fetch	high	>=3.1.1	node-fetch is vulnerable to Exposure of Sensitive Information to an Unauthorized Actor	https://github.com/advisories/GHSA-r683-j2x4-v87g	@nestjs/core>@nuxtjs/opencollective>node-fetch

### Minimum reproduction code

-

### Steps to reproduce

1. npm i
  **message**: 
  found 9 high severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
2. npm audit
     **message** :
     High            node-fetch is vulnerable to Exposure of Sensitive             
                  Information to an Unauthorized Actor                          

  Package         node-fetch                                                    

  Patched in      >=3.1.1                                                       

  Dependency of   7cbe6f22abc4b54af7c8e3d7b4de18f70b2a6ca596ebcf3243f7de2c7fc…  

  Path            7cbe6f22abc4b54af7c8e3d7b4de18f70b2a6ca596ebcf3243f7de2c7fc…  
                  >                                                             
                  7d2f6aa1cf2988150f4b2a53942778071e6e8f4fc512fb93e4b075de77c…  
                  >                                                             
                  7dbc5c2f8c28530cd0cccc14db4466b29506f0d1ca7496780bd9a54f8a2…  
                  >                                                             
                  7cbe6f22abc4b54af7c8e3d7b4de18f70b2a6ca596ebcf3243f7de2c7fc…  
                  >                                                             
                  7d2f6aa1cf2988150f4b2a53942778071e6e8f4fc512fb93e4b075de77c…  
                  >                                                             
                  7dbc5c2f8c28530cd0cccc14db4466b29506f0d1ca7496780bd9a54f8a2…  
                  >                                                             
                  3ccdad27d510c11f8e38b7a4aab8abf406b9e75051f21ae682a6c6efbac…  
                  >                                                             
                  7836f168f41f00b3a1c39f1853c48f940922be30854fa4f119a6e2356a4…  
                  > @nestjs/core > @nuxtjs/opencollective > node-fetch          

                 More info       https://github.com/advisories/GHSA-r683-j2x4-v87g        

### Expected behavior

The nestjs/core depend on node-fetch is 2.6.7 ,but node-fetch latest is 3.2.0. Why not upgrade 3.x?

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [X] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [X] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [X] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

8.2.3

### Packages versions

```json
"@nestjs/axios": "^0.0.3",
    "@nestjs/common": "^8.2.3",
    "@nestjs/config": "^1.1.5",
    "@nestjs/core": "^8.2.3",
    "@nestjs/microservices": "^8.2.3",
    "@nestjs/platform-fastify": "^8.2.3",
    "@nestjs/swagger": "^5.1.5",
    "@nestjs/terminus": "^8.0.3",
    "@nestjs/typeorm": "^8.0.2",
    "cache-manager": "^3.6.0",
    "cache-manager-ioredis": "^2.1.0",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.13.2",
    "fastify-swagger": "^4.12.6",
    "ioredis": "^4.28.2",
    "mysql2": "^2.3.3",
    "nats": "^2.4.0",
    "nestjs-pino": "^2.3.1",
    "pino": "^7.5.1",
    "pino-http": "^6.4.0",
    "rxjs": "^7.4.0",
    "tslib": "^2.3.1",
    "typeorm": "^0.2.41"
```

### Node.js version

14.17.4

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
> The `@nestjs/core` depend on `node-fetch`

actually, like you've show: `@nestjs/core > @nuxtjs/opencollective > node-fetch`, nestjs doesn't depends on `node-fetch` directly.

>  Why not upgrade node-fetch@3.x?

I guess `@nuxtjs/opencollective` didn't use v3 due to this

![image](https://user-images.githubusercontent.com/13461315/150635082-3029ed37-0b29-4e4b-8d14-67cfe6714553.png)

Also, `0.3.2` is the latest version of `@nuxtjs/opencollective`

```
$ npm view @nuxtjs/opencollective version
0.3.2
```

https://github.com/nestjs/nest/blob/6ffdef14bc9e4a77d8b4b20a302fd864f31c9124/packages/core/package.json#L30

**Comment by micalevisk** [comment]
btw if you run `npm install --ignore-scripts` your machine wouldn't get affected by this vulnerability as `@nuxtjs/opencollective` gets evaluated only in this life-cycle script:

https://github.com/nestjs/nest/blob/8673b8286312026c919260468810cc31d0ae4be1/packages/core/package.json#L19-L20

---

I'm wondering if would be that bad if `@nuxtjs/opencollective` gets moved to dev. dep., and then `@nestjs/cores` could use [`pinst`](https://www.npmjs.com/package/pinst) to run `postinstall` hook only in development :thinking:

**Comment by micalevisk** [comment]
@nadjin could you please report this to https://github.com/nuxt-contrib/opencollective

**Comment by micalevisk** [comment]
another ideia: you can use [NPM overrides](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) to get ride of that package on our side using [this GitHub repo](https://github.com/micalevisk/stub-opencollective) :smile: 

https://user-images.githubusercontent.com/13461315/150651542-e9f25b7e-b3d4-4c44-8a2a-1bd3fd49c507.mp4

**Comment by farshadmb** [comment]
hi, that might be a bit tricky however, this solution would fix your problem. 
just replace `"node-fetch": "^2.6.1"` to `"node-fetch": "^3.2.0"` then enter `npm install`. 

this forces the npm to fetch the right node-fetch. also consider as a dependency version for all packages is dependent.

**Comment by farshadmb** [comment]
Also, I've come out with a new one. 
```Bash
$ npm i node-fetch@latest
``` 
then modify `package.json`
```JSON
"overrides": {
    "@nuxtjs/opencollective@*": {
      "node-fetch@*": {
        "node-fetch": "$node-fetch"
      }
    }
  }
  ```

**Comment by micalevisk** [comment]
this was fixed in [`node-fetch@2.6.7`](https://github.com/advisories/GHSA-r683-j2x4-v87g) as well. And the latest version of `@nestjs/core` (8.2.6) doesn't have such vulnerability anymore.

![image](https://user-images.githubusercontent.com/13461315/153777348-4fc46675-5ec2-41ea-9da7-2769297c3c7c.png)
